mpi-proxy-split/lower-half/copy_stack.c

```diff
@@ -84,7 +84,7 @@ char *deepCopyStack(int argc, char **argv, char *argc_ptr, char *argv_ptr,
                         32 /* max.  padding is 32 for x86_64 */ +
                         auxv_size + env_ptr_size + argv_ptr_size +
                         (auxv_size + env_ptr_size) % 16 /* 16-byte aligned */ +
```

Comment:
Aha!  I fell into a trap that you prepared for me.  :-)

You have a commit with msg:
>Fix bug when kernel loader copy stack
And you have a commit with msg:
> Fix bug that kernel loader uses wrong argc addr

Can you please combine the two commits, since it's two fixes for a `single` piece of code.

----
**This is my old comment from the first commit.  Please ignore it:**


As we discussed, argv_ptr - argc_ptr should have been small.  It's large because `_start()` copied argc to a second address, which is passed to `main()`.  But the second address doesn't exist when we load (i.e., kernel-loader).

So, let's at least add a comment to that effect.  But even better, we can copy some of kernel-loader-v12.tgz (copy-stack.c) here:
```
  // The fnc _start() may copy 'argc', but this is its original address.
  char *argc_ptr_original = argv_ptr - sizeof(char *);
  // This shows that for current stack (not the copied one), _start copied argc.
  assert(*(long int *)argc_ptr_original == *(long int *)argc_ptr);
  ...
              (argv_ptr - argc_ptr_original); 
```